engine/src/main/java/org/terasology/persistence/typeHandling/gson/LegacyGsonTypeHandlerAdapter.java

```diff
@@ -32,6 +32,7 @@
  * object, can be used to (de)serialize objects to JSON (via Gson) with the rules specified by
  * the {@link #typeHandler}.
  */
+@Deprecated
```

Comment:
Tiny best practice recommendation for the future: leave a `@deprecated` inside the javadoc as well explaining why :-)